processor/src/main/java/org/mapstruct/ap/internal/model/IterableMappingMethod.java

```diff
@@ -379,6 +385,15 @@ else if ( other.selectionParameters != null ) {
             return false;
         }
 
+        if ( this.factoryMethod != null ) {
```

Comment:
I thought about it when I wrote the code. Its perhaps more clear. At the other hand, I let the IDE do its work. I reckon its better at "not making errors" than I am :smile:. I can change it however. WDYT?